### DISCUSSION TITLE
Problem with library list while compiling

### DISCUSSION THREAD

**Discussion by F1r3BlaZe** [codefori/vscode-ibmi#1886] [Q&A] [answered]
Hi, i've just started using code for ibm i so maybe I am doing something wrong but i have the following issue.

I built my library list with chglibl libl(**_RAWKUNDR10_** RAW01R10 R10QAB027 R10QAB026 R10QAB025 .....)
The source file i want to compile is in the library "RAWKUNDR10". There is also my SRVPGM which is causing the problem.

If I now start the action CRTBNDRPG it shows the library list in reversed order which leads to the wrong/old source file from my SRVPGM.
I could turn the library list around and the compilation will work but if I open the source with ctrl + click while coding it opens the old version of the source file from the wrong library.

**current library list:**
![grafik](https://github.com/codefori/vscode-ibmi/assets/161619756/f3f4fdbb-9f72-4aae-a200-90b07e4154b9)

**compilation output:**
Running Action: Create Bound RPG Program (CRTBNDRPG) (16:43:01)
Current library: QGPL
Library list: R10QAB025 R10QAB026 R10QAB027 RAW01R10 **_RAWKUNDR10_** SB400CONV

Is this working as intended or is my use of the library list wrong?

**Comment by worksofliam** [comment, marked as answer]
Your use is correct and this is a bug. It will be fixed in 2.8.0, but you can switch to the prerelease for now inside of VS Code on the Code for IBM i extension page
